.github/workflows/build_and_test.yml

```diff
@@ -1838,7 +1838,7 @@ jobs:
       if: steps.cache-artifact.outputs.cache-hit != 'true'
       run: |
         cd OpenDDS
-        ./configure --compiler=g++ --no-debug --features=uses_wchar=1 --std=c++03 --tests --ace=$GITHUB_WORKSPACE/OpenDDS/ACE_TAO/ACE --mpc=$GITHUB_WORKSPACE/MPC
+        ./configure --compiler=gcc --no-debug --features=uses_wchar=1 --std=c++03 --tests --ace=$GITHUB_WORKSPACE/OpenDDS/ACE_TAO/ACE --mpc=$GITHUB_WORKSPACE/MPC
```

Comment:
`gcc` is a C compiler, so I think this will break the build. What I meant was that the `update-alternatives` command should be changed to `g++`, but I guess I didn't make that clear.

Reply:
Why do we need `update-alternatives`?

Reply:
I don't know if we do. I was just trying to work with what's already here. I'm guessing @simpsont-oci had a reason for using it?

Reply:
Wasn't me! Then again, I didn't make any of the jobs that use non-default compilers (except ASAN, but that's still just the default version of clang).

Reply:
Sorry, I assumed since it was here and seemed like like frequent pattern, you wrote it. Looking at the actual git history I see now that Clayton added them.

Reply:
I'm not sure what the objection to update-alternatives is? I am using it to specify which version of gcc/g++/clang I want to be used for the build.  Doing prints of gcc --version or g++ --version show that it is using the one I specified.

Reply:
CI testing should ensure that however the user specifies the compiler, the combination of `configure` and CMake needs to handle it.  In order to test that this works, the test cases should be more than just the system's default compiler

This seems to be outside the scope of the current PR so maybe just note it and move on.

Reply:
Another con of `update-alternatives`, at least in Ubuntu 18.04, is that `CMAKE_CXX_COMPILER` is `/usr/bin/c++` by default, which would have to be to set in `update-alternatives`. We could make `update-alternatives` work by setting `c++` and `g++`/`clang++`, but it'd be more complicated than just passing `clang++-X` or `g++-X` to `configure` and `cmake`.